### ISSUE TITLE
Padded-blocks false positive with decorators

### ISSUE THREAD

**Issue by stremlenye** [babel/babel-eslint#151]
Eslint version: `v1.0.0-rc-1`
message: `Block must not be padded by blank lines  padded-blocks`

Code:
```js
function foo () {
  @decorator
  class A {

  }
  return A
}
```

Rule:
```json
"parser": "babel-eslint"
…
"padded-blocks": [2, "never"]
```
Version:
```json
"babel-eslint": "^3.1.23"
```
[Related link from eslint issue](https://github.com/eslint/eslint/issues/3097)

**Comment by hzoo** [comment]
Yeah I don't think we can do anything here - probably should make an issue in https://github.com/babel/eslint-plugin-babel for this?

The block starts at line 1.  The first node (the class) starts at line 3 (even though the decorator is there at line 2)) so the rule checks if 1+2 <= 3.

```js
function isBlockTopPadded(node) {
        var blockStart = node.loc.start.line,
            first = node.body[0],
            firstLine = first.loc.start.line,
            expectedFirstLine = blockStart + 2,
```

**Comment by jquense** [comment]
I think the plugin is probably the place to handle this, tho I am a bit unsure if there is anyway to garner this from the AST (can I visit decorators, maybe just CallExpressions?). In any case yeah it seems like another rule to add to the plugin collection

**Comment by hzoo** [comment]
Yeah you should be able to visit decorators - https://github.com/babel/babel/blob/7fa684b12076d7f4539d83acde23e9be460c6e7f/packages/babel/src/types/definitions/experimental.js and it looks like there was a rule related to decorators although not in the same way `babel/new-cap`

Going to close for now in favor of making a new rule in the plugin (can reopen if you guys think otherwise)

**Comment by stremlenye** [comment]
Thanks!
I'll open an issue at plugin repo.
